File: gestion-ressources-ng22/src/app/core/models/affectation-dto.ts
import { PersonelDto } from '.././personel-dto';
import { ProduitDto } from "./produit-dto";

export class AffectationDto {
  id: number= 0;
  quantite: string= "";
   date: string="";
   motif: string="";
   produitDto: ProduitDto= new ProduitDto();
   personelDto: PersonelDto= new PersonelDto();
   
}

File: backend/src/main/java/com/odji/spring_back_end/affectation/dto/AffectationDto.java
package com.odji.spring_back_end.affectation.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AffectationDto {

    private Integer id;

    private BigDecimal quantite;

    private LocalDate date;

    private String motif;

    // Relations sortantes exposées (sans le suffixe "Dto")
    private ProduitDto produit;
    private PersonelDto personel;
}
